delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/slowqueries/BlazegraphSlowQueryStore.scala

```diff
@@ -0,0 +1,55 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.slowqueries
+
+import ch.epfl.bluebrain.nexus.delta.kernel.database.Transactors
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.slowqueries.model.BlazegraphSlowQuery
+import ch.epfl.bluebrain.nexus.delta.sdk.views.ViewRef
+import doobie.implicits._
+import doobie.postgres.implicits._
+import ch.epfl.bluebrain.nexus.delta.sourcing.implicits._
+import io.circe.syntax.EncoderOps
+import monix.bio.Task
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Identity.Database._
+
+import java.time.Instant
+
+trait BlazegraphSlowQuerySink {
+  def save(query: BlazegraphSlowQuery): Task[Unit]
+}
+
+trait BlazegraphSlowQueryStore extends BlazegraphSlowQuerySink {
```

Comment:
What is the value to separate the save operation ? 
It makes you create two different instances and complicates things even more.

My guess is partly to do the test where we want to write in database and it fails.
I would rather see this check go for the reason that if postgres is down, we don't bother about logging a slow query because we won't even be able to perform the query at all (we can't fetch the acls, we can't fetch the view, etc...)

Reply:
Only one instance is created. This interface segregation removes boilerplate from the tests

The postgres check is a feature we could add, but I don't know if it's a valuable optimisation

If postgres is down I'm not sure the query would be slow enough to be logged anyway